### ISSUE TITLE
Postgres deploy in UK South fails

### ISSUE THREAD

**Issue by racingferret** [terraform-providers/terraform-provider-azurerm#1023]
### Terraform Version
Terraform v0.11.5 (also tested with v0.11.3)
+ provider.azurerm v1.3.0

### Affected Resource(s)
- azurerm_postgresql_server

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
resource "azurerm_resource_group" "test" {
  name     = "azure-postgres-test"
  location = "UK South"
}

resource "azurerm_postgresql_server" "test" {
  name                = "testserver"
  location            = "${azurerm_resource_group.test.location}"
  resource_group_name = "${azurerm_resource_group.test.name}"

  sku {
    name     = "PGSQLB50"
    capacity = "50"
    tier     = "Basic"
  }

  administrator_login          = "pgadmin"
  administrator_login_password = "pgpassword"
  version                      = "9.6"
  storage_mb                   = "51200"
  ssl_enforcement              = "Disabled"
}

resource "azurerm_postgresql_database" "pgsql_database" {
  name                = "testdb"
  resource_group_name = "${azurerm_resource_group.test.name}"
  server_name         = "${azurerm_postgresql_server.test.name}"
  charset             = "UFT8"
  collation           = "en-GB"
}

resource "azurerm_postgresql_firewall_rule" "pgsql_fw_rule" {
  name                = "office"
  resource_group_name = "${azurerm_resource_group.test.name}"
  server_name         = "${azurerm_postgresql_server.test.name}"
  start_ip_address    = "10.0.0.1"
  end_ip_address      = "10.0.0.1"
}
```
### Expected Behavior
Postgres database should be created without error in the UK South region

### Actual Behavior
Error: Error applying plan:

1 error(s) occurred:

* azurerm_postgresql_server.test: 1 error(s) occurred:

* azurerm_postgresql_server.test: Long running operation terminated with status 'Failed': Code="ProvisioningDisabled" Message="Gen4 hardware is not supported for this region."

### Steps to Reproduce
1. `terraform apply`

**Comment by Andre-Freitas** [comment]
i also have the same problem in Central US. [Azure Postgresql compute generations](https://docs.microsoft.com/en-us/azure/postgresql/concepts-pricing-tiers#compute-generations-vcores-and-memory)

**Comment by tombuildsstuff** [comment]
hi @racingferret @Andre-Freitas 

Thanks for opening this issue :)

Taking a look into this, I believe this issue should be resolved by #948, which will expose the new pricing schema/SKU's to Terraform and allow either Gen4/Gen5 hardware to be used depending on which region/SKU you're targeting. Rather than having multiple issues tracking the same thing, I'm going to close this issue in favour of #948 

Thanks!
